DataSetExplorer/DataSetBuilder/IDataSetCreationService.cs

```diff
@@ -1,11 +1,14 @@
-﻿using DataSetExplorer.DataSetSerializer.ViewModel;
+﻿using DataSetExplorer.DataSetBuilder.Model;
+using DataSetExplorer.DataSetSerializer.ViewModel;
 using FluentResults;
 
 namespace DataSetExplorer
 {
     public interface IDataSetCreationService
     {
+        public Result<DataSet> CreateDataSetInDatabase(string basePath, string projectName, string projectAndCommitUrl);
         public Result<string> CreateDataSetSpreadsheet(string basePath, string projectName, string projectAndCommitUrl);
-        public Result<string> CreateDataSetSpreadsheet(string basePath, string projectName, string projectAndCommitUrl, NewSpreadSheetColumnModel columnModel);
+        public Result<string> CreateDataSetSpreadsheet(string basePath, string projectName, NewSpreadSheetColumnModel columnModel, DataSet dataSet);
+        public Result<DataSet> GetDataSetIfCreated(int id);
```

Comment:
"GetDataset" is sufficient - it will be created, but the question is what is it's status, and this is something the front-end will deal with.